selenium_tests/course_info/course_info_tests.py

```diff
@@ -0,0 +1,106 @@
+        """
+        # try to load the base url, but expect to be interrupted by the pin
+        # login page
+        pin_page = PinLoginPageObject(self.driver)
+        pin_page.login(self.BASE_URL, self.USERNAME, self.PASSWORD)
+        edit_page = EditorPage(self.driver)
+        self.assertTrue(edit_page.is_loaded(), 'edit page not loaded')
+        edit_page.focus_on_default_content()
+
+        # click on our tool's button in the editor, verify the tool's field
+        # selection page comes up in the "modal" iframe
+        edit_page.tool_button.click()
+        edit_page.focus_on_tool_frame()
+        WebDriverWait(self.driver, 30).until(
+            lambda d: edit_page.registrar_code_checkbox.is_displayed(),
+            'Timed out waiting for registrar code checkbox to be displayed')
+
+        # just for completeness sake, let's verify that *all* checkboxes are there
+        checkbox_properties = [p for p in edit_page.located_properties
+                                   if p.endswith('_checkbox')]
+        for checkbox in checkbox_properties:
+            self.assertTrue(getattr(edit_page, checkbox).is_displayed(),
+                            '{} is not displayed'.format(checkbox))
+
+
+    def test_insert_widget_all_fields(self):
+        """
+        Tests inserting the widget into a page, leaving all fields selected.
+        """
+        # deal with the login
+        pin_page = PinLoginPageObject(self.driver)
+        pin_page.login(self.BASE_URL, self.USERNAME, self.PASSWORD)
+        edit_page = EditorPage(self.driver)
+
+        # delete any instances of the widget already in the page
+        edit_page.focus_on_default_content()
+        edit_page.focus_on_editor_frame()
+        for widget in edit_page.get_inserted_widgets():
+            widget.click()
+            edit_page.editor_body.send_keys(Keys.DELETE)
+
+        # now squawk if any are left
+        edit_page.focus_on_default_content()
+        edit_page.focus_on_editor_frame()
+        self.assertEqual(edit_page.get_inserted_widgets(), [],
+                         'Found unexpected instances of widget already in the page')
+
+        # launch the tool, click save in the tool iframe
+        edit_page.focus_on_default_content()
+        edit_page.tool_button.click()
+        edit_page.focus_on_tool_frame()
+        edit_page.tool_submit_button.click()
+
+        # verify the iframe is gone
+        edit_page.focus_on_default_content()
+        with self.assertRaises(TimeoutException):
+            WebDriverWait(self.driver, 10).until(
+                lambda d: d.find_element(*edit_page.locator_class.REGISTRAR_CODE_CHECKBOX).is_visible())
+
+        # now make sure the widget is in our editor frame
+        edit_page.focus_on_editor_frame()
+        widgets = edit_page.get_inserted_widgets()
+        self.assertEqual(len(widgets), 1)
+
+        # figure out which f values we should be expecting
+        expected_f_values = {
```

Comment:
Can we make this chunk a `PageObject` function, to encapsulate explicit references to its locator_class, etc?